core/src/main/java/com/datastax/oss/driver/internal/core/cql/Conversions.java

```diff
@@ -441,4 +450,77 @@ public static CoordinatorException toThrowable(
         return new ProtocolError(node, "Unknown error code: " + errorMessage.code);
     }
   }
+
+  /** Returns a common size for all kinds of Request implementations. */
+  static int minimumRequestSize(Request statement, DriverContext context) {
+    Preconditions.checkArgument(
+        context instanceof InternalDriverContext,
+        "DriverContext provided cannot be used to calculate statement's size");
+
+    /* Header and payload are common inside a Frame at the protocol level */
+
+    // Frame header has a fixed size of 9 for protocol version > V3, which includes Frame flags size
+    int size = headerEncodedSize();
+
+    if (!statement.getCustomPayload().isEmpty()
+        && ((InternalDriverContext) context)
+            .protocolVersionRegistry()
+            .supports(context.protocolVersion(), ProtocolFeature.CUSTOM_PAYLOAD)) {
+      size += PrimitiveSizes.sizeOfBytesMap(statement.getCustomPayload());
+    }
+
+    /* These are options in the protocol inside a frame that are common to all Statements */
+
+    size += queryFlagsSize(context.protocolVersion().getCode());
+
+    size += PrimitiveSizes.SHORT; // size of consistency level
+    size += PrimitiveSizes.SHORT; // size of serial consistency level
+    size += PrimitiveSizes.LONG; // size of default timestamp
+
+    return size;
+  }
+
+  /**
+   * Returns the size in bytes of a simple statement's values, depending on whether the values are
+   * named or positional.
+   */
+  static int sizeOfSimpleStatementValues(
+      SimpleStatement simpleStatement,
+      ProtocolVersion protocolVersion,
+      CodecRegistry codecRegistry) {
+    int size = 0;
+    if (!simpleStatement.getPositionalValues().isEmpty()) {
+      int totalValuesSize =
+          Values.sizeOfPositionalValues(
+              simpleStatement
+                  .getPositionalValues()
+                  .stream()
```

Comment:
In general we are trying to avoid using streams if possible (reasoning outlined [here](https://github.com/datastax/java-driver/blob/4.x/CONTRIBUTING.md#no-stream-api)).

Reply:
Hm ok. Will modify to follow the guidelines.